pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -317,6 +316,35 @@ func (r *ReconcileClusterDeployment) Reconcile(ctx context.Context, request reco
 		return reconcile.Result{}, nil
 	}
 
+	// Remove legacy conditions if present:
+	legacyConditions := []string{
+		"ClusterImageSetNotFound",
+	}
+	newConditions = []hivev1.ClusterDeploymentCondition{}
+	for _, c := range cd.Status.Conditions {
+		legacy := false
+		for _, lc := range legacyConditions {
+			if string(c.Type) == lc {
+				legacy = true
+				break
+			}
+			if !legacy {
+				newConditions = append(newConditions, c)
+			} else {
+				cdLog.Infof("removing legacy condition: %v", c)
+			}
+		}
+	}
```

Comment:
Can't we use `FindClusterDeploymentCondition`? Looks like it is working [here](https://github.com/openshift/hive/pull/1430/files#diff-b9685aa0bf796d1db13613fe411ed287146777107405bcc935ebd1d45d64ea47R1457)

Reply:
That would let me find if it's there, but we have no delete function and probably shouldn't given we want them always there (except for legacy). Reslicing looked a little clunk, and requires an index, which FindClusterDeploymentCondition doesn't expose.  Given all this it felt more reasonable to just iterate the slice and built up a new with the ones we want to keep. Let me know if I missed something obvious though.

Reply:
The logic is good - I just meant to see if we could make it not clunky - it is fine this way...